sysutils/apcupsd/Portfile

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem                  1.0
+
+name                        apcupsd
+version                     3.14.14
+categories                  sysutils
+platforms                   darwin
+license                     GPL-2
+maintainers                 stephenreay
+description                 APC UPS Daemon
+long_description            Apcupsd can be used for power mangement and controlling most of APC’s UPS models on Unix and Windows machines.
+homepage                    http://www.apcupsd.org
+master_sites                https://jaist.dl.sourceforge.net/project/apcupsd/apcupsd%20-%20Stable/${version}/
+checksums                   rmd160  ecab483f1a38d1cfb75c6d439f89a611efcc45f2 \
+                            sha256  db7748559b6b4c3784f9856561ef6ac6199ef7bd019b3edcd7e0a647bf8f9867 \
+                            size    1843409
+
+configure.args              --prefix=${prefix} \
+                            --sbindir=${prefix}/sbin \
+                            --sysconfdir=${prefix}/etc/apcupsd \
+                            ac_cv_func_which_gethostbyname_r=no
+
+patchfiles                  apcagent-makefile.diff \
+                            darwin-makefile.diff \
+                            apcupsd-plist.diff
+
+startupitem.autostart       true
+startupitem.install         true
+startupitem.custom_file     ${worksrcpath}/platforms/darwin/org.apcupsd.apcupsd.plist
+startupitem.type            launchd
+
+variant usb description {Add USB support} {
+    configure.args-append   --enable-usb
+
+    depends_lib-append      port:libusb-compat
+
+    post-activate {
+        system "ln -sf ${prefix}/Library/Extensions/ApcupsdDummy.kext /Library/Extensions/"
+        system "kmutil load -p /Library/Extensions/ApcupsdDummy.kext; true"
+        system "echo 'APC requires a Kernel Extension for USB connections.'"
+        system "open /System/Library/PreferencePanes/Security.prefPane"
+    }
+}
+
+variant modbus description {Add MODBUS support} {
+    configure.args-append   --enable-modbus-usb
```

Comment:
You probably want to add "--disable-modbus-usb" as a default argument to avoid it being found if it is installed (i.e., opportunistic linking). Then here you can do a "configure.args-replace".